cli/azd/internal/appdetect/appdetect.go

```diff
@@ -280,21 +280,12 @@ type walkDirFunc func(path string, entries []fs.DirEntry) error
 // walkDirectories walks the file tree rooted at root, calling fn for each directory in the tree, including root.
 // The directories are walked in lexical order.
 func walkDirectories(root string, fn walkDirFunc) error {
-	info, err := os.Lstat(root)
-	if err != nil {
-		return err
-	}
-
-	return walkDirRecursive(root, fs.FileInfoToDirEntry(info), fn)
-}
-
-func walkDirRecursive(path string, d fs.DirEntry, fn walkDirFunc) error {
-	entries, err := os.ReadDir(path)
+	entries, err := os.ReadDir(root)
```

Comment:
@weikanglim - Just drawing your attention to this one - I don't think these changes have any semantic difference, but I figured you might want a quick once over of this after realizing that we were ignoring a parameter.  Maybe we should be making a larger change here because we expected a different semantics.

Reply:
The `fs.DirEntry` was addendum metadata along with the `path` being visited, it looks fine. I just pushed a small tweak to documentation and naming to reflect the newly combined walking function.